pkg/registry/common/connect/nse_server.go

```diff
@@ -18,93 +18,183 @@ package connect
 
 import (
 	"context"
-	"time"
-
-	"github.com/networkservicemesh/sdk/pkg/tools/clienturlctx"
-	"github.com/networkservicemesh/sdk/pkg/tools/clock"
-
-	"github.com/networkservicemesh/sdk/pkg/registry/common/clienturl"
-	"github.com/networkservicemesh/sdk/pkg/tools/extend"
+	"net/url"
 
 	"github.com/golang/protobuf/ptypes/empty"
 	"github.com/networkservicemesh/api/pkg/api/registry"
+	"github.com/pkg/errors"
 	"google.golang.org/grpc"
 
 	"github.com/networkservicemesh/sdk/pkg/registry/core/adapters"
+	"github.com/networkservicemesh/sdk/pkg/tools/clienturlctx"
+	"github.com/networkservicemesh/sdk/pkg/tools/multiexecutor"
 )
 
-type nseCacheEntry struct {
-	expirationTimer clock.Timer
-	client          registry.NetworkServiceEndpointRegistryClient
-}
+// NSEClientFactory is a NSE client chain supplier func type
+type NSEClientFactory func(ctx context.Context, cc grpc.ClientConnInterface) registry.NetworkServiceEndpointRegistryClient
```

Comment:
Let's try do not to create additional types to prevent make the system complex.

```suggestion
type NSEClientFactory = func(ctx context.Context, cc grpc.ClientConnInterface) registry.NetworkServiceEndpointRegistryClient
```

Reply:
Please apply this for similar places.

Reply:
Yep, done.